### PR TITLE
CDNC-2088

### PR DESCRIPTION
**Pull request by bowenxia**
**What changed?**
Add AttemptCount into log

**Why?**
Currently we have metrics for number-of-attempts and have alerts on them, but finding the associated workflows can be tricky due to other logs also existing / temporary issues that self-recover happening all the time.

**How did you test it?**
Unit test

**Potential risks**
no

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: service/history/task/task.go [hunk 1]
```diff
@@ -236,10 +236,12 @@ func (t *taskImpl) HandleErr(
 			t.attempt++
 			if t.attempt > t.criticalRetryCount() {
 				t.scope.RecordTimer(metrics.TaskAttemptTimerPerDomain, time.Duration(t.attempt))
-				t.logger.Error("Critical error processing task, retrying.",
+				t.logger.Error("Critical error processing task, retrying.", // TODO: there might be more. If there is, do the same thing
 					tag.Error(err),
 					tag.OperationCritical,
+					// fixme!!! Add a new tag attempt into TaskType.log the attempt
```
Comment (neil-xie) [inline service/history/task/task.go:242]: It is better to create a new ticket for it and link to your current ticket or just fix it if it is a simple fix.

Comment (bowenxia) [reply service/history/task/task.go:242]: Ohh I forgot to delete those comments. Those were reminders for myself. Thanks Neil.

Comment (bowenxia) [reply service/history/task/task.go]: The comments were deleted.

**Comment by neil-xie** [comment]
from your note, do you need to add the tag here https://github.com/uber/cadence/blob/master/service/history/task/cross_cluster_task.go#L362

**Comment by bowenxia** [comment]
@neil-xie Yes you are right! Nice catch. I totally missed that one. I've searched all other uses of the metric TaskAttemptTimerPerDomain, and made sure that everything was covered.

**Review by neil-xie — APPROVED** [review]
lg

**Review by davidporter-id-au — APPROVED** [review]
nice!

**Pull request merged**